applications/ShapeOptimizationApplication/python_scripts/algorithm_trust_region.py

```diff
@@ -0,0 +1,682 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Baumgaertner Daniel, https://github.com/dbaumgaertner
+#
+# ==============================================================================
+
+# Making KratosMultiphysics backward compatible with python 2.6 and 2.7
+from __future__ import print_function, absolute_import, division
+
+# Kratos Core and Apps
+from KratosMultiphysics import *
+from KratosMultiphysics.ShapeOptimizationApplication import *
+
+# Additional imports
+from algorithm_base import OptimizationAlgorithm
+from custom_math import NormInf3D, Dot, ScalarVectorProduct, Norm2, RowSize, HorzCat, Minus, Plus, Trans, Prod, ZeroVector, ZeroMatrix, IsEmpty, SafeConvertVectorToMatrix
+from custom_math import TranslateToNewBasis, TranslateToOriginalBasis, QuadProg, PerformBisectioning, SolveLinearSystem
+from custom_variable_utilities import WriteDictionaryDataOnNodalVariable, ReadNodalVariableToList, WriteNodeCoordinatesToList, WriteListToNodalVariable
+from custom_timer import Timer
+import mapper_factory
+import data_logger_factory
+import copy
+
+# ==============================================================================
+class AlgorithmTrustRegion(OptimizationAlgorithm):
```

Comment:
a docstring would be nice

Reply:
will follow once the algorithm is settled and tested a while.